central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -186,3 +186,34 @@ func getScopeKeys(scanClusters []*storage.ComplianceOperatorScanConfigurationV2_
 
 	return clusterScopeKeys
 }
+
+func (ds *datastoreImpl) RemoveClusterFromScanConfig(ctx context.Context, clusterID string) error {
+	q := search.NewQueryBuilder().
+		AddExactMatches(search.ClusterID, clusterID).ProtoQuery()
+	scans, err := ds.GetScanConfigurations(ctx, q)
+	if err != nil {
+		return err
+	}
+	for _, scan := range scans {
+		clusters := scan.GetClusters()
+		newClusters := []*storage.ComplianceOperatorScanConfigurationV2_Cluster{}
+		for _, cluster := range clusters {
+			if cluster.GetClusterId() != clusterID {
+				newClusters = append(newClusters, cluster)
+			}
+
+		}
+		scan.Clusters = newClusters
+		err := ds.UpsertScanConfiguration(ctx, scan)
```

Comment:
I don't know how likely this is but there's a possibility of a race condition here. 

Req 1: gets scan config (line 193)
Req 2: updates scan config to include another cluster (or some other detail).
Req 1: upserts _old_ config back in. 

Since we don't acquire row locks (AFAIK) this might be a bit difficult to fix against. Perhaps another check just to make sure no other configurations have changed on the scan config?

Thoughts @dashrews78 @rhmdnd  ?